lib/subjects/selector.rb

```diff
@@ -128,9 +128,15 @@ def find_subject_queue
     end
 
     def dequeue_ids(queue, sms_ids)
+      background_update = !queue_user
       if queue_user
-        queue.dequeue_update(sms_ids)
-      else
+        begin
+          queue.dequeue_update(sms_ids)
+        rescue ActiveRecord::StaleObjectError
+          background_update = true
```

Comment:
I wonder if having more duplication would be more readable in this particular case, by not having to keep track of state in your head (the idea being following if-else branches is easier than remembering variable values).

```ruby
if queue_user
  begin
    queue.dequeue_update(sms_ids)
  rescue ActiveRecord::StaleObjectError
    DequeueSubjectQueueWorker.perform_async(queue.id, sms_ids)
  end
else
  DequeueSubjectQueueWorker.perform_async(queue.id, sms_ids)
end
```